webui/src/lib/auth/authContext.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { createContext, useContext, useMemo, useState, ReactNode } from "react";
+import { AUTH_STATUS, type AuthStatus } from "./status";
+
+type AuthContextShape = {
+    status: AuthStatus;
+    markAuthenticated: () => void;
+    markUnauthenticated: () => void;
+};
+
+const AuthContext = createContext<AuthContextShape | null>(null);
+const STORAGE_KEY = "lakefs:ui:auth:status";
+
+function readPersistedStatus(): AuthStatus {
+    try {
+        const v = window.localStorage.getItem(STORAGE_KEY);
```

Comment:
AFAIK, `localStorage` is not secured, and can be accessed by other apps.
Can you describe the implications in this case? Are the risks mitigated somehow?